digdag-standards/src/main/java/io/digdag/standards/operator/aws/EmrOperatorFactory.java

```diff
@@ -0,0 +1,1753 @@
+
+            Aws.configureProxy(emrClientConfiguration, emrEndpoint, environment);
+            Aws.configureProxy(s3ClientConfiguration, s3Endpoint, environment);
+            Aws.configureProxy(kmsClientConfiguration, kmsEndpoint, environment);
+
+            AmazonElasticMapReduce emr = new AmazonElasticMapReduceClient(credentials, emrClientConfiguration);
+            AmazonS3Client s3 = new AmazonS3Client(credentials, s3ClientConfiguration);
+            AWSKMSClient kms = new AWSKMSClient(credentials, kmsClientConfiguration);
+
+            // Set up file stager
+            Optional<AmazonS3URI> staging = params.getOptional("staging", String.class).transform(s -> {
+                try {
+                    return new AmazonS3URI(s);
+                }
+                catch (IllegalArgumentException ex) {
+                    throw new ConfigException("Invalid staging uri: '" + s + "'", ex);
+                }
+            });
+            Filer filer = new Filer(s3, staging, workspace, templateEngine, params);
+
+            // TODO: make it possible for operators to _reliably_ clean up
+            boolean cleanup = false;
+
+            try {
+                TaskResult result = run(tag, emr, kms, filer);
+                cleanup = true;
+                return result;
+            }
+            catch (Throwable t) {
+                boolean retry = t instanceof TaskExecutionException &&
+                        ((TaskExecutionException) t).getRetryInterval().isPresent();
+                cleanup = !retry;
+                throw Throwables.propagate(t);
+            }
+            finally {
+                if (cleanup) {
+                    // Best effort clean up of staging
+                    try {
+                        filer.tryCleanup();
+                    }
+                    catch (Throwable t) {
+                        logger.warn("Failed to clean up staging: {}", staging, t);
+                    }
+                }
+                s3.shutdown();
+                emr.shutdown();
+            }
+        }
+
+        private AWSCredentials credentials(String tag)
+        {
+            SecretProvider awsSecrets = context.getSecrets().getSecrets("aws");
+            SecretProvider emrSecrets = awsSecrets.getSecrets("emr");
+
+            String accessKeyId = emrSecrets.getSecretOptional("access_key_id")
+                    .or(() -> awsSecrets.getSecret("access_key_id"));
+
+            String secretAccessKey = emrSecrets.getSecretOptional("secret_access_key")
+                    .or(() -> awsSecrets.getSecret("secret_access_key"));
+
+            AWSCredentials credentials = new BasicAWSCredentials(accessKeyId, secretAccessKey);
+
+            Optional<String> roleArn = emrSecrets.getSecretOptional("role_arn")
+                    .or(awsSecrets.getSecretOptional("role_arn"));
+
+            if (!roleArn.isPresent()) {
+                return credentials;
```

Comment:
Okay. Using BasicAWSCredentials here would be enough although I think we can reduce security risk using a temporary credential just in case when the original credentials is leaked.